### PR TITLE
Optional inputs

### PR DESCRIPTION
**Pull request by sgoley**
Took a first pass at solving my own feature request: https://github.com/fal-ai/dbt-cloud-action/issues/14

Attempts to add the following new inputs to the action's request body to the dbt api: 
 * schema_override
 * target_name_override
 
 Source API docs here: https://docs.getdbt.com/dbt-cloud/api-v2#tag/Jobs/operation/triggerRun

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -17,6 +17,14 @@ inputs:
...
     required: true
     default: "Triggered by a GitHub Action"
+  schema_override:
+    description: "Override the destination schema in the configured target for this job."
+    required: false
+    default: "dbt_cloud_pr_123" 
+  target_name_override: 
+     description: "Override the target.name context variable used when running this job"
+     required: false
+     default: "staging"
```
Comment (chamini2) [inline action.yml]: I would *not* set a default value for these and let dbt handle that.

**Review by chamini2 — CHANGES_REQUESTED** [review]
Thanks for contributing. What do you think if instead we just let the user set the whole object instead of adding each optional property to the optional inputs?

```yaml
      - uses: fal-ai/dbt-cloud-action@main
        id: dbt_cloud_run
        with:
          dbt_cloud_token: ${{ secrets.DBT_CLOUD_API_TOKEN }}
          dbt_cloud_account_id: ${{ secrets.DBT_CLOUD_ACCOUNT_ID }}
          dbt_cloud_job_id: ${{ secrets.DBT_CLOUD_JOB_ID }}
          trigger_run_body:
            cause: "My cause"
            schema_override: ${{ secrets.schema }}
```

And we would just have to have a default value for that object's `cause` property before sending the request.

File: action.yml (hunk 1 above)
Comment (goley-aveone) [reply action.yml:27]: By that you mean to drop the whole line, not just empty string right?

Comment (sgoley) [reply action.yml]: By that you mean to remove the whole default line 27, not just empty string correct?

**Comment by sgoley** [comment]
> Thanks for contributing. What do you think if instead we just let the user set the whole object instead of adding each optional property to the optional inputs?
> 
> ```yaml
>       - uses: fal-ai/dbt-cloud-action@main
>         id: dbt_cloud_run
>         with:
>           dbt_cloud_token: ${{ secrets.DBT_CLOUD_API_TOKEN }}
>           dbt_cloud_account_id: ${{ secrets.DBT_CLOUD_ACCOUNT_ID }}
>           dbt_cloud_job_id: ${{ secrets.DBT_CLOUD_JOB_ID }}
>           trigger_run_body:
>             cause: "My cause"
>             schema_override: ${{ secrets.schema }}
> ```
> 
> And we would just have to have a default value for that object's `cause` property before sending the request.

I would say the reason not to make the whole schema a single string in secrets is so that values can be set dynamically like this: 

https://gist.github.com/sgoley/af3e1f857251786afb1d739ada2e2a03#file-fmtdeploy-yml-L83

Not sure if thats what you meant or not.

**Comment by sgoley** [comment]
Actually realized today that to mirror the dbtcloud native intregation, we will also need include the `git_branch` param.

Do you know if its possible to set the input default for that to an environment variable like `${{ github.head_ref }}`
(name of the branch which kicked off the action?) 

Something like: 

``` yml
  git_branch:
     description: "The git branch to check out before running this job"
     required: false
     default: ${{ github.head_ref }}
```

**Comment by chamini2** [comment]
This is now supported, check [v4.0](https://github.com/fal-ai/dbt-cloud-action/releases/tag/v4.0) release for details.

**Pull request closed without merging**